packages/gatsby/src/utils/dev-ssr/render-dev-html.ts

```diff
@@ -49,8 +49,13 @@ export const restartWorker = (htmlComponentRendererPath): void => {
 
 const searchFileForString = (substring, filePath): Promise<boolean> =>
   new Promise(resolve => {
+    const escapedSubString = substring.replace(/[.*+?^${}()|[\]\\]/g, `\\$&`)
```

Comment:
this was causing failing tests on the development-runtime for paths like `-[client-only]` as they'd crash the server